### PR TITLE
Run batch_processing if building master

### PR DESCRIPTION
**Pull request by Drulex**
Partially addresses #2888.

### CODE REVIEW

**Comment by Drulex** [comment]
I guess we'll need to merge to see make sure it happens. Assigning @kousu for sanity check.

**Comment by Drulex** [comment]
Hmm looks like we can't ask `git` for the branch.. https://unhashable.com/getting-the-current-branch-name-during-a-pull-request-in-travis-ci/

File: .ci.sh [hunk 1]
```diff
@@ -30,14 +30,7 @@ echo *** ANALYZE CODE ***
...
-# pip install sphinx sphinxcontrib.programoutput sphinx_rtd_theme
-# cd documentation/sphinx
-# make html
-# cd -
-
-# python create_package.py -s ${TRAVIS_OS_NAME}  # test package creation
-# cd ../spinalcordtoolbox_v*
-# yes | ./install_sct  # test installation of package
-
+if [[ "$BRANCH" == "master" ]]; then
```
Comment (kousu) [inline .ci.sh:33]: Maybe `$TRAVIS_BRANCH` to be more explicit? :shrug:

File: .ci.sh [hunk 2]
```diff
@@ -30,14 +30,7 @@ echo *** ANALYZE CODE ***
...
+  echo "*** Running batch_processing.sh ***"
+  ./batch_processing.sh
+fi
```
Comment (kousu) [inline .ci.sh:36]: I was hoping that ./ci.sh could be more agnostic than this.

In https://github.com/neuropoly/spinalcordtoolbox/issues/2674 I mused about adopting the pattern in https://docs.travis-ci.com/user/speeding-up-the-build/#parallelizing-your-builds-across-virtual-machines of

```
env:
  - SUITE=integration_tests
  - SUITE=unit_tests
  - SUITE=lint
  - SUITE=super_integration_tests # i.e. batch_processing.sh

...

script: "./travis.sh $SUITE"
```

Looking now I'm not confident I understand how `env` interacts with `matrix: include: ...`. We probably want batch_processing.sh to run on the same cases as `if: branch = release or type = cron`, but would want `SUITE=lint` to only run on one VM (say, Ubuntu). Is there a good way to build that kind of cross-product in Travis on top of what we already have in there?

**Review by kousu — APPROVED** [review]
Take a look at my thoughts, but this seems fine and if you want to merge as is go for it.

File: .ci.sh (hunk 1 above)
Comment (Drulex) [reply .ci.sh:33]: `TRAVIS_BRANCH` is actually something else. See (https://docs.travis-ci.com/user/environment-variables/#Default-Environment-Variables). Relevant section:
image

So based on this logic, when testing PRs, `TRAVIS_BRANCH` == target branch == `master`. This is the reason for the little gymnastics in `.ci.sh` to figure out if we are in a PR or not and use the proper env var.

File: .ci.sh (hunk 2 above)
Comment (Drulex) [reply .ci.sh:36]: I agree, I also investigated parallelization, though I investigated the "distribute on multiple cores" approach (use of stages) vs the "distribute on multiple vm" approach that you are proposing (see https://github.com/neuropoly/spinalcordtoolbox/issues/2888#issuecomment-689585541).

Your approach would be more environment agnostic which I think is a good idea. I propose that we continue that investigation in a separate issue (created #2923 ).

**Pull request merged**